src/vllm_tgis_adapter/__main__.py

```diff
@@ -54,14 +54,15 @@ async def override_signal_handler() -> None:
             for sig in (signal.SIGINT, signal.SIGTERM):
                 loop.add_signal_handler(sig, signal_handler)
 
-        tasks.append(loop.create_task(override_signal_handler()))
+        await override_signal_handler()
```

Comment:
Given the way that signal handlers are set in `serve_http` [here](https://github.com/opendatahub-io/vllm/blob/main/vllm/entrypoints/launcher.py#L29-L42), I think this will be overridden.

This is not an issue though: we can remove both `signal_handler()` and `override_signal_handler()`.

In this case ctrl-C will be handled by the http server, stopping the server task on ctrl-c. Then `asyncio.wait(tasks, return_when=FIRST_COMPLETED)` will return and both tasks will be cancelled in the `for task in tasks` loop below.

Here's the patch:
```diff
diff --git a/src/vllm_tgis_adapter/__main__.py b/src/vllm_tgis_adapter/__main__.py
index 8021dde..139f91b 100644
--- a/src/vllm_tgis_adapter/__main__.py
+++ b/src/vllm_tgis_adapter/__main__.py
@@ -43,26 +43,9 @@ async def start_servers(args: argparse.Namespace) -> None:
         )
         tasks.append(grpc_server_task)
 
-        def signal_handler() -> None:
-            # prevents the uvicorn signal handler to exit early
-            for task in tasks:
-                task.cancel()
-
-        async def override_signal_handler() -> None:
-            loop = asyncio.get_running_loop()
-
-            for sig in (signal.SIGINT, signal.SIGTERM):
-                loop.add_signal_handler(sig, signal_handler)
-
-        await override_signal_handler()
-
         with contextlib.suppress(asyncio.CancelledError):
-            # Both server tasks will exit normally on shutdown, so we await
-            # FIRST_COMPLETED to catch either one shutting down.
-            await asyncio.wait(
-                tasks,
-                return_when=FIRST_COMPLETED,
-            )
+            await asyncio.wait(tasks, return_when=FIRST_COMPLETED)
+
         for task in tasks:
             task.cancel()
 ```

Reply:
oh yeah, you're right. More negative diff!